app/controllers/concerns/foreman/controller/provisioning_templates.rb

```diff
@@ -42,4 +43,17 @@ def process_template_kind
   def type_name_singular
     @type_name_singular ||= resource_class.to_s.underscore
   end
+
+  def global_registration_vars
+    permitted = Foreman::Plugin.all
+                               .map(&:allowed_registration_vars)
+                               .flatten.compact.uniq
+    {
+      user: User.current,
+      auth_token: User.current.jwt_token!(expiration: 4.hours.to_i),
+      organization: Organization.authorized(:view_organizations).find_by(id: params['organization_id']),
```

Comment:
If the org ID is not found, does this raise an exception or does it return nil? (same question for location and hostgroup)

Reply:
Fixed.
If params are present and resource is not found, Foreman will generate not found error.